src/typing/type_inference_js.ml

```diff
@@ -5219,10 +5223,10 @@ and mk_type_params cx ?(map=SMap.empty) (types: Ast.Identifier.t list) =
   let typeparams, smap =
     List.fold_left mk_type_param ([], SMap.empty) types
   in
-  List.rev typeparams, smap
+  List.rev typeparams, SMap.union smap type_params_map
```

Comment:
Always return the old type params with the new ones mixed in. Before, some call sites would do this, but this seems cleaner, assuming it's sound.